Messages/Message.cs

```diff
@@ -23,24 +23,22 @@ namespace Messages
      // Also need decorator
      // Tweak however works
 
+    //Base-concrete message class
     public class Message{
```

Comment:
I had imagined a message decorator as a member of the message class. Getting the concept down, let me know if I'm wrong. The message class is a pure virtual class. The Message decorator inherits from Message and provides functionality for the message class. The Individual sub-messages can add things or override. Because of the properties of inheritance we can grab any message as MessageDecorator, decode the type, and then set the type? Or if we call decode will the entire message be decoded?

Reply:
This MessageDecorator class allows us to stack on the extra member variables from a Decorating class.
With the override of gameid and conversation id, it is just making sure its keeping the existing id's when the message was instantiated.
The Decorating classes (I.E. CreateGameMessage, StartGameMessage,...etc), all abstract from a MessageDecorator. This is key because it allows us to stack the extra member variables and extra responsibilities. 
Think of the MessageDecorator Class as a middle man that allows us to stack extra responsibilities. 

**Example of usage:**
Message msg = new Message();
msg = new CreateGameDecorator(){ DecoratedMessage = msg, MinPlayer = 2....etc };
- Now `msg` is a CreateGame Message that we can now send via UDP/TCP

Reply:
In order for the decorator pattern to work correctly, `MessageDecorator` needs to be a separate class that abstracts from `Message`. `MessageDecorator` holds a `Message` object as a member variable noting that this is the thing it is decorating.

Reply:
oh, okay, that's really nice. Then we can have a function that assigns id's a standard way and still have unique messages. Thanks for clarifying.